File: backend-nexus/app/schemas/landing.py
"""Schemas for the tenant landing page API - single response with all data."""

from datetime import datetime
from pydantic import BaseModel
from typing import Optional


class TenantPublicCard(BaseModel):
    """Minimal tenant info for public listing (active tenants only)."""

    id: int
    slug: Optional[str] = None
    name: str
    moto: Optional[str] = None
    about_text: Optional[str] = None
    logo: Optional[str] = None
    image: Optional[str] = None
    brand_color_primary: Optional[str] = None
    brand_color_secondary: Optional[str] = None
    brand_color_background: Optional[str] = None
    brand_color_foreground: Optional[str] = None
    brand_color_muted: Optional[str] = None

    class Config:
        from_attributes = True


class ServiceLandingItem(BaseModel):
    id: int
    name: str
    price: float
    description: Optional[str] = None
    is_active: bool = True

    class Config:
        from_attributes = True


class DepartmentLandingItem(BaseModel):
    id: int
    name: str
    phone_number: Optional[str] = None
    email: Optional[str] = None
    location: Optional[str] = None
    services: list[ServiceLandingItem] = []

    class Config:
        from_attributes = True


class DoctorLandingItem(BaseModel):
    user_id: int
    first_name: str
    last_name: str
    specialization: Optional[str] = None
    education: Optional[str] = None
    licence_number: Optional[str] = None
    is_active: bool = True
    working_hours: Optional[dict] = None

    class Config:
        from_attributes = True


class ProductLandingItem(BaseModel):
    product_id: int
    tenant_id: int
    name: str
    description: Optional[str] = None
    category: Optional[str] = None
    image_url: Optional[str] = None
    price: float
    stock_quantity: int = 0
    is_available: bool = True

    class Config:
        from_attributes = True


class TenantLandingRead(BaseModel):
    id: int
    name: str
    slug: Optional[str] = None
    email: str
    licence_number: str

    class Config:
        from_attributes = True


class TenantDetailsLandingRead(BaseModel):
    tenant_id: int
    logo: Optional[str] = None
    image: Optional[str] = None
    moto: Optional[str] = None
    brand_color_primary: Optional[str] = None
    brand_color_secondary: Optional[str] = None
    brand_color_background: Optional[str] = None
    brand_color_foreground: Optional[str] = None
    brand_color_muted: Optional[str] = None
    title: Optional[str] = None
    about_text: Optional[str] = None
    font_id: Optional[int] = None
    font_name: Optional[str] = None
    font_header_family: Optional[str] = None
    font_body_family: Optional[str] = None

    class Config:
        from_attributes = True


class PlanLandingItem(BaseModel):
    """Active plan shown on the public tenant landing page."""

    id: int
    name: str
    description: Optional[str] = None
    price: float
    duration: Optional[int] = None
    max_appointments: Optional[int] = None
    max_consultations: Optional[int] = None
    is_active: bool = True

    class Config:
        from_attributes = True


class TenantLandingPageResponse(BaseModel):
    tenant: TenantLandingRead
    details: Optional[TenantDetailsLandingRead] = None
    departments: list[DepartmentLandingItem] = []
    doctors: list[DoctorLandingItem] = []
    products: list[ProductLandingItem] = []
    plans: list[PlanLandingItem] = []

File: frontend-nexus/src/interfaces/landing.ts
/**
 * Types for tenant landing and public tenant list (match backend schemas/landing.py)
 */

export interface TenantPublicCard {
  id: number;
  slug: string | null;
  name: string;
  moto: string | null;
  about_text: string | null;
  logo: string | null;
  image: string | null;
  brand_color_primary: string | null;
  brand_color_secondary: string | null;
  brand_color_background: string | null;
  brand_color_foreground: string | null;
  brand_color_muted: string | null;
}

export interface ServiceLandingItem {
  id: number;
  name: string;
  price: number;
  description: string | null;
  is_active: boolean;
}

export interface DepartmentLandingItem {
  id: number;
  name: string;
  phone_number: string | null;
  email: string | null;
  location: string | null;
  services: ServiceLandingItem[];
}

export interface TenantLandingRead {
  id: number;
  name: string;
  slug: string | null;
  email: string;
  licence_number: string;
}

export interface DoctorLandingItem {
  user_id: number;
  first_name: string;
  last_name: string;
  specialization: string | null;
  education: string | null;
  licence_number: string | null;
  is_active: boolean;
  working_hours: Record<string, unknown> | null;
}

export interface ProductLandingItem {
  product_id: number;
  tenant_id: number;
  name: string;
  description: string | null;
  category: string | null;
  image_url: string | null;
  price: number;
  stock_quantity: number;
  is_available: boolean;
}

export interface TenantDetailsLandingRead {
  tenant_id: number;
  logo: string | null;
  image: string | null;
  moto: string | null;
  brand_color_primary: string | null;
  brand_color_secondary: string | null;
  brand_color_background: string | null;
  brand_color_foreground: string | null;
  brand_color_muted: string | null;
  title: string | null;
  slogan: string | null;
  about_text: string | null;
  font_key: string | null;
  font_id: number | null;
  font_name: string | null;
  font_header_family: string | null;
  font_body_family: string | null;
}

export interface PlanLandingItem {
  id: number;
  name: string;
  description: string | null;
  price: number;
  duration: number | null;
  max_appointments: number | null;
  max_consultations: number | null;
  is_active: boolean;
}

export interface TenantLandingPageResponse {
  tenant: TenantLandingRead;
  details: TenantDetailsLandingRead | null;
  departments: DepartmentLandingItem[];
  doctors: DoctorLandingItem[];
  products: ProductLandingItem[];
  plans: PlanLandingItem[];
}
